Support re exporting

crate_b re exports as:
```rust
pub use rust_decimal;
pub use rust_decimal_macros;
```
crate_a uses macro:
```rust
use crate_b::rust_decimal::Decimal;
use crate_b::rust_decimal_macros::dec;
let a: Decimal = dec!(1.1);
```
We get: 
```
error[E0433]: failed to resolve: could not find `rust_decimal` in `{{root}}`
```
Is there a reason why we are calling `::rust_decimal::Decimal` ?

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -94,3 +94,9 @@ jobs:
         with:
           command: make
           args: test-serde
+
+      - name: Run macro tests
+        uses: actions-rs/cargo@v1
+        with:
+          command: make
+          args: test-macros

--- a/Makefile.toml
+++ b/Makefile.toml
@@ -40,7 +40,8 @@ dependencies = [
     "test-maths",
     "test-misc",
     "test-db",
-    "test-serde"
+    "test-serde",
+    "test-macros"
 ]
 
 [tasks.test-db]
@@ -58,6 +59,9 @@ dependencies = [
     "test-serde-arbitrary-precision",
     "test-serde-arbitrary-precision-float"
 ]
+
+[tasks.test-macros]
+workspace = true
 
 [tasks.test-no-std]
 command = "cargo"

--- a/macros/Cargo.toml
+++ b/macros/Cargo.toml
@@ -15,5 +15,9 @@ license = "MIT"
 rust_decimal = { path = "..", version = "1.10.3" }
 quote = "1.0"
 
+[features]
+default = []
+reexportable = []
+
 [lib]
 proc-macro = true

--- a/macros/Makefile.toml
+++ b/macros/Makefile.toml
@@ -1,0 +1,13 @@
+[tasks.test-macros]
+dependencies = [
+"test-macros-default",
+"test-macros-reexportable"
+]
+
+[tasks.test-macros-default]
+command = "cargo"
+args = ["test", "--tests"]
+
+[tasks.test-macros-reexportable]
+command = "cargo"
+args = ["test", "--tests", "--features=reexportable"]

--- a/macros/src/lib.rs
+++ b/macros/src/lib.rs
@@ -1,3 +1,38 @@
+//!
+//! A helpful macro for instantiating `Decimal` numbers.
+//!
+//! By default, this requires `rust_decimal` to be available at the project root. e.g. the macro
+//! will effectively produce:
+//!
+//! ```ignore
+//! ::rust_decimal::Decimal::from_parts(12345, 0, 0, false, 4)
+//! ```
+//!
+//! While this is convenient for most use cases, it is sometimes not desired behavior when looking
+//! to reexport the libray. Consequently, this behavior can be modified by enabling the feature
+//! `reexportable`. When this feature is enabled, the macro will instead reproduce the functional
+//! equivalent of:
+//!
+//! ```ignore
+//! Decimal::from_parts(12345, 0, 0, false, 4)
+//! ```
+//!
+//! # Examples
+//!
+//! ```rust
+//! use rust_decimal_macros::dec;
+//!
+//! // If the reexportable feature is enabled, `Decimal` needs to be in scope
+//! #[cfg(feature = "reexportable")]
+//! use rust_decimal::Decimal;
+//!
+//! let number = dec!(1.2345);
+//! assert_eq!("1.2345", number.to_string());
+//! let number = dec!(-5.4321);
+//! assert_eq!("-5.4321", number.to_string());
+//! ```
+//!
+
 use core::str::FromStr;
 use proc_macro::TokenStream;
 use quote::quote;
@@ -19,14 +54,27 @@ pub fn dec(input: TokenStream) -> TokenStream {
     };
 
     let unpacked = decimal.unpack();
-    // We need to further unpack these for quote for now
-    let lo = unpacked.lo;
-    let mid = unpacked.mid;
-    let hi = unpacked.hi;
-    let negative = unpacked.is_negative;
-    let scale = unpacked.scale;
+    expand(
+        unpacked.lo,
+        unpacked.mid,
+        unpacked.hi,
+        unpacked.is_negative,
+        unpacked.scale,
+    )
+}
+
+#[cfg(not(feature = "reexportable"))]
+fn expand(lo: u32, mid: u32, hi: u32, negative: bool, scale: u32) -> TokenStream {
     let expanded = quote! {
         ::rust_decimal::Decimal::from_parts(#lo, #mid, #hi, #negative, #scale)
+    };
+    expanded.into()
+}
+
+#[cfg(feature = "reexportable")]
+fn expand(lo: u32, mid: u32, hi: u32, negative: bool, scale: u32) -> TokenStream {
+    let expanded = quote! {
+        Decimal::from_parts(#lo, #mid, #hi, #negative, #scale)
     };
     expanded.into()
 }

--- a/macros/tests/macro_tests.rs
+++ b/macros/tests/macro_tests.rs
@@ -1,5 +1,9 @@
 use rust_decimal_macros::dec;
 
+// Require using for reexportable feature
+#[cfg(feature = "reexportable")]
+use rust_decimal::Decimal;
+
 #[test]
 fn it_can_parse_decimal() {
     let tests = &[

